scripts/create_synthetic_population.R

```diff
@@ -0,0 +1,95 @@
+## =================================#
+## Setup ---------------
+## =================================#
+library(tidyverse)
+library(tigris)
+library(tidycensus)
+library(patchwork)
+
+set.seed(1234)
+
+state_synth <- "WY"
+year_synth <- 2023
+population_size <- 1000
+
+## =================================#
+## Get population ---------------
+## =================================#
+pums_vars <- pums_variables |>
+  filter(year == 2018, survey == "acs1") |>
+  distinct(var_code, var_label, data_type, level)
+
+person_variables <- c("SPORDER", "SERIALNO", "PWGTP", "AGEP", "SEX", "PUMA", "REGION")
+house_variables <- c("WGTP", "NP")
+
+sample_pums <- get_pums(
+  variables = c(person_variables, house_variables),
+  state = state_synth,
+  survey = "acs1",
+  year = year_synth
+)
+
+household_pums <- sample_pums |>
+  dplyr::select(SERIALNO, all_of(house_variables)) |>
+  distinct()
+
+## =================================#
+## Create population ---------------
+## =================================#
+synth_pop_df <- tibble()
+house_counter <- 0
+while (nrow(synth_pop_df) < population_size) {
+  house_counter <- house_counter + 1
+  synth_pop_df <- synth_pop_df |>
+    bind_rows(household_pums |>
+      sample_n(1, weight = WGTP) |>
+      left_join(sample_pums, by = (c("SERIALNO", "WGTP", "NP"))) |>
+      mutate(house_number = house_counter))
+}
+
+## =================================#
+## Recode and math GEO -----------
+## =================================#
+## For now, we will use PUMA codes
+## instead of census tracts
+pumas_st <- pumas(state = state_synth)
+tracts_st <- tracts(state = state_synth)
+
+synth_pop_region_df <- synth_pop_df |>
+  left_join(
+    pumas_st |>
+      dplyr::select(STATEFP20, PUMACE20, INTPTLAT20, INTPTLON20),
+    by = c("PUMA" = "PUMACE20")
+  ) |>
+  dplyr::select(-geometry) |>
+  mutate(
+    region_id = sprintf("%02d%09d", as.numeric(STATE), as.numeric(PUMA)),
```

Comment:
Even though region ID isn't in the ASPR set, so we wouldn't name it in camel case for consistency across implementations, should it be in camel case to stylistically match `homeId`?

Reply:
You're right